File: problems/problems_2028/solution.go
package problem2028

import (
	"encoding/json"
	"log"
	"strings"
)

func missingRolls(rolls []int, mean int, n int) []int {
	remain := mean * (len(rolls) + n)
	for _, roll := range rolls {
		remain -= roll
	}
	if remain < n || remain > 6*n {
		return nil
	}
	avg, extra := remain/n, remain%n
	ans := make([]int, n)
	for i := 0; i < extra; i++ {
		ans[i] = avg + 1
	}
	for i := extra; i < n; i++ {
		ans[i] = avg
	}
	return ans
}

func Solve(input string) any {
	values := strings.Split(input, "\n")
	var rolls []int
	var mean int
	var n int

	if err := json.Unmarshal([]byte(values[0]), &rolls); err != nil {
		log.Fatal(err)
	}
	if err := json.Unmarshal([]byte(values[1]), &mean); err != nil {
		log.Fatal(err)
	}
	if err := json.Unmarshal([]byte(values[2]), &n); err != nil {
		log.Fatal(err)
	}

	return missingRolls(rolls, mean, n)
}

File: problems/problems_2028/Solution.java
package problems.problems_2028;

import com.alibaba.fastjson.JSON;
import java.util.*;
import qubhjava.BaseSolution;


public class Solution extends BaseSolution {
    public int[] missingRolls(int[] rolls, int mean, int n) {
        int remain = mean * (rolls.length + n);
        for (int roll: rolls) {
            remain -= roll;
        }
        if (remain < n || remain > n * 6) {
            return new int[]{};
        }
        int avg = remain / n, extra = remain % n;
        int[] ans = new int[n];
        for (int i = 0; i < n; i++) {
            ans[i] = avg + (i < extra ? 1 : 0);
        }
        return ans;
    }

    @Override
    public Object solve(String[] values) {
        int[] rolls = jsonArrayToIntArray(values[0]);
		int mean = Integer.parseInt(values[1]);
		int n = Integer.parseInt(values[2]);
        return JSON.toJSON(missingRolls(rolls, mean, n));
    }
}
